### PR TITLE
install google-cloud-sdk in ubuntu image

### LINKED ISSUE #16: Install gcloud SDK in the build images
It looks that when logged Docker is configured to use GCP credentials (i.e. `gcloud auth activate-service-account`, `gcloud auth configure-docker`), then Bazel needs `docker-credential-gcloud` from gcloud SDK to be available in `$PATH` in order to pull Docker images `container_pull`.

### PR DESCRIPTION
**Pull request by lizan**
Fixes #16 

Signed-off-by: Lizan Zhou <lizan@tetrate.io>

### CODE REVIEW

File: build_container/build_container_ubuntu.sh [hunk 1]
```diff
@@ -18,6 +18,12 @@ update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 1000
 update-alternatives --config gcc
 update-alternatives --config g++
 
+# Google Cloud SDK
+echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] http://packages.cloud.google.com/apt cloud-sdk main" \
+  | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | \
```
Comment (htuch) [inline build_container/build_container_ubuntu.sh]: It might be nice to pin here, WDYT?

Comment (lizan) [reply build_container/build_container_ubuntu.sh]: do you mean pinning gcloud to a specific version? it is possible but I don't see much value to do it for now. We can always revisit that later if we see some issue for this picking up specific version of gcloud.

Comment (htuch) [reply build_container/build_container_ubuntu.sh]: No, I mean SHA256 of the GPG key.

Comment (lizan) [reply build_container/build_container_ubuntu.sh]: I believe the gpg key is periodically updated so it is not a good idea to pin either. This is from https://cloud.google.com/sdk/docs/quickstart-debian-ubuntu and it mentions about expired key.

**Pull request merged**